manipulation/src/main/java/org/jboss/gm/manipulation/actions/UploadTaskTransformerAction.java

```diff
@@ -0,0 +1,29 @@
+package org.jboss.gm.manipulation.actions;
+
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.maven.MavenResolver;
+import org.gradle.api.tasks.Upload;
+import org.jboss.gm.common.alignment.AlignmentModel;
+
+/**
+ * Fixes pom.xml generation in old "maven" plugin.
+ * <p>
+ * Adds PomTransformer to all MavenResolver repositories in Upload tasks.
+ */
+public class UploadTaskTransformerAction implements Action<Project> {
+
+    private AlignmentModel.Module alignmentConfiguration;
+
+    public UploadTaskTransformerAction(AlignmentModel.Module alignmentConfiguration) {
+        this.alignmentConfiguration = alignmentConfiguration;
+    }
+
+    @Override
+    public void execute(Project project) {
+        project.getTasks().withType(Upload.class).all(upload -> upload.getRepositories()
+                .withType(MavenResolver.class).all(resolver -> {
+                    resolver.getPom().withXml(new PomTransformer(alignmentConfiguration));
```

Comment:
I am wondering if this is the best approach (same for `PublicationPomTransformerAction`)... We could also do ` resolver.getPom().whenConfigured(...)` and then work with the `MavenPom` type which would be much easier than playing with XML. WDYT?

Reply:
Actually it seems like the snippet I pasted above would only work for `UploadTaskTransformerAction` and not for `PublicationPomTransformerAction`

Reply:
We can certainly re-visit things as we see fit.